examples/sep24/sep24.ts

```diff
@@ -182,6 +215,27 @@ export const runWithdrawWatcher = (anchor, kp) => {
   stop = resp.stop;
 };
 
+const walletSigner = DefaultSigner;
+walletSigner.signWithDomainAccount = async ({
+  transactionXDR,
+  networkPassphrase,
+  accountKp,
+}: Types.SignWithDomainAccountParams): Promise<Transaction> => {
+  if (!clientSecret) {
+    throw new Error("Client Secret missing from .env file");
+  }
+
+  const signer = SigningKeypair.fromSecret(clientSecret);
+
+  const transaction = TransactionBuilder.fromXDR(
+    transactionXDR,
+    networkPassphrase,
+  ) as Transaction;
+  signer.sign(transaction);
+
+  return Promise.resolve(transaction);
```

Comment:
just curious - what's the benefit of making this function `async`? Everything within this function runs synchronously and then we return a resolved Promise. We're not actually waiting for anything async to resolve

Reply:
ah good question. It's because in practice the `signWithDomainAccount` method is for signing the transaction by a backend server, see [here](https://github.com/stellar/typescript-wallet-sdk/blob/a97ea11976d1b08bd47fb2c46b562e3a2667b17d/src/walletSdk/Auth/WalletSigner.ts#L74-L86)
(or at least giving the user the option to sign it w/ a server)

but here we're signing with a .env secret for the example, so doesn't have to be async but making it so to follow the function type